src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -89,34 +88,60 @@
       BuildWithOnlineSources="$(BuildWithOnlineSources)"
       SourceBuildSources="$(SourceBuildSources)" />
 
-    <WriteLinesToFile File="$(RepoCompletedSemaphorePath)UpdateNuGetConfig.complete" Overwrite="true" />
+    <MakeDir Directories="$(BaseIntermediateOutputPath)" />
+    <Touch Files="$(BaseIntermediateOutputPath)UpdateNuGetConfig.complete" AlwaysCreate="true">
+      <Output TaskParameter="TouchedFiles" ItemName="FileWrites" />
+    </Touch>
   </Target>
 
+  <!-- Update the SDK version in the repo's global.json file.
+       This guarantees that all repositories build with the VMR's SDK version. -->
   <Target Name="UpdateGlobalJsonVersions"
           BeforeTargets="Build"
           Condition="'$(GlobalJsonFile)' != ''"
-          Inputs="$(MSBuildProjectFullPath)"
-          Outputs="$(RepoCompletedSemaphorePath)UpdateGlobalJsonVersions.complete">
+          Inputs="$(MSBuildProjectFullPath);$(MSBuildThisFileFullPath)"
+          Outputs="$(BaseIntermediateOutputPath)UpdateGlobalJsonVersions.complete">
     <ItemGroup>
-      <_PossibleCliVersionJsonPath Include="sdk.version" />
-      <_PossibleCliVersionJsonPath Include="tools.dotnet" />
+      <_PossibleCliVersionJsonPath Include="sdk:version" />
+      <_PossibleCliVersionJsonPath Include="tools:dotnet" />
     </ItemGroup>
 
     <UpdateJson JsonFilePath="$(GlobalJsonFile)"
                 PathToAttribute="%(_PossibleCliVersionJsonPath.Identity)"
-                NewAttributeValue="$(SDK_VERSION)"
+                NewAttributeValue="$(NETCoreSdkVersion)"
+                SkipUpdateIfMissingKey="true" />
+
+    <!-- VB PoC, remove the global runtimes until https://github.com/dotnet/arcade/issues/14283 is resolved. -->
+    <UpdateJson JsonFilePath="$(GlobalJsonFile)"
+                PathToAttribute="tools:runtimes"
                 SkipUpdateIfMissingKey="true" />
 
-    <WriteLinesToFile File="$(RepoCompletedSemaphorePath)UpdateGlobalJsonVersions.complete" Overwrite="true" />
+    <MakeDir Directories="$(BaseIntermediateOutputPath)" />
+    <Touch Files="$(BaseIntermediateOutputPath)UpdateGlobalJsonVersions.complete" AlwaysCreate="true">
+      <Output TaskParameter="TouchedFiles" ItemName="FileWrites" />
+    </Touch>
+  </Target>
+
+  <!-- TODO: Remove when all repos use a consistent set of eng/common files: https://github.com/dotnet/source-build/issues/3710. -->
+  <Target Name="UpdateEngCommonFiles"
+          Condition="'$(RepositoryName)' != 'arcade' and ('$(UpdateEngCommonFiles)' == 'true' or '$(DotNetBuildVertical)' == 'true')"
```

Comment:
Is UpdateEngCommonFiles set anywhere?

Reply:
No, this is a dev innerloop switch in case you want to overwrite eng/common files for source-build as well locally.